crates/next-core/src/nodejs/node_rendered_source.rs

```diff
@@ -141,73 +145,69 @@ impl ContentSource for NodeRenderContentSource {
-                                                anyhow!("url needs to be provided")
-                                            })?,
-                                            query: query.clone(),
-                                            headers: headers.clone(),
-                                            path: format!("/{path}"),
-                                        }
-                                        .cell(),
-                                    )
-                                    .into(),
-                                )
-                                .cell());
+                        let entry = this.entry.entry(data.clone()).await?;
+                        let asset = render_static(
+                            this.server_root.join(path),
+                            entry.module,
+                            this.runtime_entries,
+                            this.fallback_page,
+                            entry.chunking_context,
+                            entry.intermediate_output_path,
+                            RenderData {
+                                params,
+                                method: data
+                                    .method
+                                    .clone()
+                                    .ok_or_else(|| anyhow!("method needs to be provided"))?,
+                                url: data
+                                    .url
+                                    .clone()
+                                    .ok_or_else(|| anyhow!("url needs to be provided"))?,
+                                query: data
+                                    .query
+                                    .clone()
+                                    .ok_or_else(|| anyhow!("query needs to be provided"))?,
+                                headers: data
+                                    .headers
+                                    .clone()
+                                    .ok_or_else(|| anyhow!("headers needs to be provided"))?,
+                                path: format!("/{path}"),
                             }
-                        }
-                        return Ok(ContentSourceResult::NeedData {
-                            source: self_vc.into(),
-                            path: path.to_string(),
-                            vary: ContentSourceDataVary {
-                                method: true,
-                                url: true,
-                                headers: Some(ContentSourceDataFilter::All),
-                                query: Some(ContentSourceDataFilter::All),
-                                ..Default::default()
-                            },
-                        }
-                        .cell());
+                            .cell(),
+                        );
+                        break 'block ContentSourceContent::Static(asset.into()).cell();
```

Comment:
Let's have an `if/else` here. I don't think the break to block construct is necessary here, it just makes things harder to read.

```rust
let content = if data.method.is_some() && ... {
    ...
    ContentSourceContent::Static(...)
} else {
    ContentSourceContent::NeedData(...)
}
```